### PR TITLE
DF-27: Added domain table for admin with add new domain and update existing domain functionalities

### PR DESCRIPTION
**Pull request by AyushGiri190**
Added domain table for admin where admin may add domain and change the status for preexisting domain
path:
src/app/domain
src/components/domain

### CODE REVIEW

File: src/components/Domain/addDomain.tsx [hunk 1]
```diff
@@ -15,31 +19,50 @@ import {
...
 });
 
-const AddDomain: React.FC = () => {
+const AddDomain: React.FC<AddDomainProps> = ({ onAddDomain }) => {
   const [open, setOpen] = useState(false);
-  const [domains, setDomains] = useState<Domain[]>([]);
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+  const [successMessage, setSuccessMessage] = useState<string | null>(null);
+  const [domains, setDomains] = useState<Domain[]>([]); // Use Domain type here
+  const currentUserId = "user-id-123";
```
Comment (Divesh-singh) [inline src/components/Domain/addDomain.tsx:63]: why is this hardcoded here

Comment (AyushGiri190) [reply src/components/Domain/addDomain.tsx:63]: there is no navigation till now the generic code is just below it

File: src/components/Domain/addDomain.tsx [hunk 2]
```diff
@@ -48,31 +71,71 @@ const AddDomain: React.FC = () => {
...
+      status: "Active",
     },
   });
+
+  // Fetch the list of domains from the backend
   useEffect(() => {
     async function fetchDomains() {
       try {
         const response = await axiosInstance.get("/domain/all");
         setDomains(response.data.data);
```
Comment (Divesh-singh) [inline src/components/Domain/addDomain.tsx:85]: verify if you got data in response.data.data or not before setting state

File: src/components/Domain/addDomain.tsx [hunk 3]
```diff
@@ -48,31 +71,71 @@ const AddDomain: React.FC = () => {
...
     try {
-      console.log("Submitting data:", data);
-      await axiosInstance.post(`/domain/create`, data);
+      const domainDataWithUser = { ...data, createdBy: currentUserId };
+      // Post the new domain to the backend
+      const response = await axiosInstance.post(
+        `/domain/createdomain`,
+        domainDataWithUser,
+      );
+      const newDomain = response.data.data;
```
Comment (Divesh-singh) [inline src/components/Domain/addDomain.tsx:117]: verify if you got data in response.data.data or not before setting state

File: src/components/Domain/addDomain.tsx [hunk 4]
```diff
@@ -48,31 +71,71 @@ const AddDomain: React.FC = () => {
...
+      setErrorMessage(null); // Clear any previous error message
+
+      // Close the dialog after a short delay
+      setTimeout(() => {
+        setOpen(false);
+        setSuccessMessage(null);
+      }, 500);
     } catch (error) {
-      console.error("Error submitting domain:", error);
+      toast({
```
[the remark this answers is not in the document]
Comment (AyushGiri190) [reply src/components/Domain/addDomain.tsx:133]: toast would be better sir ? as it is also used elsewhere

File: src/components/Domain/domainTable.tsx [hunk 5]
```diff
@@ -23,53 +21,118 @@ import {
...
-        console.log("API Response:", response.data);
+  const [noData, setNoData] = useState(false); // State to handle no data available
+  const { toast } = useToast();
+  // Function to fetch domain data
+  const fetchDomainData = async () => {
+    setLoading(true);
+    setNoData(false); // Reset noData state before fetching
+    try {
+      const response = await axiosInstance.get("/domain/all");
+      if (response.data.data.length === 0) {
```
Comment (Divesh-singh) [inline src/components/Domain/domainTable.tsx:46]: This traversal 'response.data.data.length' is very dangerous if you don't get response in desired format verify response before traversing properties

File: src/components/Domain/domainTable.tsx [hunk 6]
```diff
@@ -23,53 +21,118 @@ import {
...
+        setNoData(true); // Set noData if response is empty
+      } else {
         setDomainData(response.data.data);
```
Comment (Divesh-singh) [inline src/components/Domain/domainTable.tsx:49]: also here, don't believe without checks that you got response in desired format

File: src/components/Domain/domainTable.tsx [hunk 7]
```diff
@@ -23,53 +21,118 @@ import {
...
+  useEffect(() => {
     fetchDomainData();
   }, []);
 
+  // Handle domain deletion
   const handleDelete = async (domainId: string) => {
-    console.log("Domain ID received in handleDelete:", domainId); // Debugging line
     if (!domainId) {
-      console.error("Domain ID is undefined.");
+      toast({
```
Comment (Divesh-singh) [inline src/components/Domain/domainTable.tsx:72]: this should be displayed if you get response from api like 404 - no domain found with this id

if you don't get domainId here simply don't proceed with api request

File: src/components/Domain/domainTable.tsx [hunk 8]
```diff
@@ -23,53 +21,118 @@ import {
...
+    setDomainData((prevData) =>
+      prevData.map((user) =>
+        user._id === labelId
+          ? { ...user, status: checked ? "Active" : "Inactive" }
+          : user,
+      ),
+    );
+    try {
+      await axiosInstance.put(`/domain/${labelId}/status`, {
+        status: checked ? "Active" : "Inactive",
```
Comment (Divesh-singh) [inline src/components/Domain/domainTable.tsx:102]: use from enum

File: src/components/Domain/domainTable.tsx [hunk 9]
```diff
@@ -78,40 +141,68 @@ const DomainTable: React.FC = () => {
...
                       <TableCell>
-                        {truncateText(domain.description, 50)}
+                        {domain.createdAt || "No Data Available"}
+                      </TableCell>
+                      <TableCell>
+                        {domain.createdBy || "No Data Available"}
+                      </TableCell>
+                      <TableCell>
+                        <Switch
+                          checked={domain.status === "Active"}
```
Comment (Divesh-singh) [inline src/components/Domain/domainTable.tsx:195]: enum

**Review by Divesh-singh — COMMENTED** [review]
Go through above comments

File: src/components/Domain/addDomain.tsx (hunk 1 above)
Comment (Divesh-singh) [reply src/components/Domain/addDomain.tsx:63]: Is there no record id recorded for current logged in user?

File: src/components/Domain/addDomain.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (Divesh-singh) [reply src/components/Domain/addDomain.tsx:133]: You have to decide what is best suited and looks cleaner for the use case, I will leave it up to you

**Review by Divesh-singh — APPROVED** [review]

**Pull request merged**